src/main/java/org/mockito/internal/util/MockUtil.java

```diff
@@ -63,13 +64,18 @@ public static TypeMockability typeMockabilityOf(Class<?> type) {
         mockMaker.resetMock(mock, newHandler, settings);
     }
 
-    public static <T> MockHandler<T> getMockHandler(T mock) {
+    public static MockHandler<?> getMockHandler(Object mock) {
         if (mock == null) {
             throw new NotAMockException("Argument should be a mock, but is null!");
         }
 
-        if (isMock(mock)) {
-            return mockMaker.getHandler(mock);
+        for (MockResolver mockResolver : Plugins.getMockResolvers()) {
+            mock = mockResolver.resolve(mock);
+        }
+
+        MockHandler handler = mockMaker.getHandler(mock);
```

Comment:
Nit: extract this out in a private method and use that below as well.

Reply:
Fully agree, I hope for some feedback first, too.